java-generator/core/src/main/java/io/fabric8/java/generator/nodes/JObject.java

```diff
@@ -75,11 +77,53 @@ public JObject(
     } else {
       String nextPackagePath = pkgPrefix + AbstractJSONSchema2Pojo.packageName(this.className);
 
+      // in order to handle duplicated fields, first let's build a map of fields grouped by their sanitized names, i.e.:
+      // 1(fieldName) -> n(fieldDefinition(key, props))
+      final Map<String, Map<String, JSONSchemaProps>> groupedFieldDefinitions = fields.entrySet().stream()
+          .collect(Collectors.groupingBy(
+              (f) -> AbstractJSONSchema2Pojo.sanitizeString(f.getKey()),
+              Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue)));
+
       for (Map.Entry<String, JSONSchemaProps> field : fields.entrySet()) {
+        String fieldKey = field.getKey();
+        // lookup the duplicated field properties map
+        final Map.Entry<String, Map<String, JSONSchemaProps>> fieldDuplicatesDefinition = groupedFieldDefinitions.entrySet()
+            .stream()
+            .filter(d -> d.getValue().containsKey(field.getKey()))
+            .findFirst().get();
+        final int duplicatesCount = fieldDuplicatesDefinition.getValue().entrySet().size();
+        if (duplicatesCount > 1) {
```

Comment:
`fieldDuplicatesDefinition` is generated from a `findFirst()` call, if I'm not missing anything this condition is not going to be triggered.
Can you add a `GeneratorTest` to cover this aspect, please 🙏 ?

Reply:
> we should be able to do a direct lookup:
> `groupedFieldDefinitions.get(AbstractJSONSchema2Pojo.sanitizeString(f.getKey()));`

Indeed, I went the long way, when instead it was just there at hand, thanks. Fixing!

> The note on testing still applies if that's not too much work ...
Sure. I'll add a couple of tests for the requirements we're enforcing: no more than 1 duplicated field && the duplicated field must be deprecated.